test/ng/directive/inputSpec.js

```diff
@@ -657,17 +657,20 @@ describe('input', function() {
       expect($rootScope.form.alias.$error.month).toBeTruthy();
     });
 
-    it('should allow four or more digits in year', function() {
-      var inputElm = helper.compileInput('<input type="month" ng-model="value"  ng-model-options="{timezone: \'UTC\'}"/>');
 
-      helper.changeInputValueTo('10123-03');
```

Comment:
AFAIK, it is indeed not possible to set the date without using the picker (unless you open up dev tools and set the `value` property directly). I guess it is a conscious choice (albeit one I'm not a big fan of :stuck_out_tongue:).